x/ccv/provider/keeper/punish_validator.go

```diff
@@ -0,0 +1,103 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	evidencetypes "github.com/cosmos/cosmos-sdk/x/evidence/types"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+)
+
+// JailAndTombstoneValidator jails and tombstones the validator with the given provider consensus address
+func (k Keeper) JailAndTombstoneValidator(ctx sdk.Context, providerAddr types.ProviderConsAddress) error {
+	validator, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.ToSdkConsAddr())
+	if !found {
+		return errorsmod.Wrapf(slashingtypes.ErrNoValidatorForAddress, "provider consensus address: %s", providerAddr.String())
+	}
+
+	if validator.IsUnbonded() {
+		return fmt.Errorf("validator is unbonded. provider consensus address: %s", providerAddr.String())
+	}
+
+	if k.slashingKeeper.IsTombstoned(ctx, providerAddr.ToSdkConsAddr()) {
+		return fmt.Errorf("validator is tombstoned. provider consensus address: %s", providerAddr.String())
+	}
+
+	// jail validator if not already
+	if !validator.IsJailed() {
+		k.stakingKeeper.Jail(ctx, providerAddr.ToSdkConsAddr())
+	}
+
+	// Jail the validator to trigger the unbonding of the validator
```

Comment:
```suggestion
```
Maybe we can remove this comment altogeter or just mention that we `JailUntil` in accordance to what `cosmos-sdk` does. The "unbonding of the validator" already happens when we `Jail` (see my other [comment](https://github.com/cosmos/interchain-security/pull/1340/files#r1356523376)).